torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1374,13 +1392,88 @@ void ONNXShapeTypeInference(
 
 } // namespace
 
+std::pair<bool, bool> IsShapeReliable(Node* n) {
+  auto reliable = true;
+  auto complete = true;
+  for (auto input : n->inputs()) {
+    reliable &=
+        ConstantValueMap::GetTypeReliable(input->debugName()).value_or(false);
+    if (auto pt = input->type()->cast<TensorType>()) {
+      if (!pt->sizes().isComplete()) {
+        complete = false;
+      }
+    }
+  }
+  return std::make_pair(reliable, complete);
+}
+
+// There is no need to put onnx type here, but we need this
+// for some legacy tests when onnx_shape_inference=False.
+static std::unordered_set<std::string> nodeTypeReliableForTracer = {
+    "prim::ListConstruct",
+    "onnx::Cast",
+    "onnx::Constant",
+    "com.microsoft::Gelu"};
```

Comment:
Do we need to increase this list when we export new contrib ops?

Reply:
It is not required. It is only needed when this new contrib op causes shape inference error && tracing result shape is reliable if all inputs are static for this contrib op type.